src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/FlowProcessingAction.java

```diff
@@ -101,4 +101,11 @@ protected boolean isRemoveCustomerPortSharedCatchRule(String flowId,
 
         return flowIds.size() == 1 && flowIds.iterator().next().equals(flowId);
     }
+
+    protected Set<String> getDiverseWithFlowIds(Flow flow) {
+        return flowRepository.findByGroupId(flow.getGroupId()).stream()
```

Comment:
Does it make sense to fetch a collection of Flow entities and then return just flow ids? Please, consider a new method in the repository.

Reply:
Fixed.